web-cmd/app.py

```diff
@@ -0,0 +1,219 @@
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(10*365*24*60*60)  # 10 years
+    cert.set_issuer(cert.get_subject())  # self signed cert
+    cert.set_pubkey(pkey)
+    cert.sign(pkey, 'sha256')
+
+    with open('key.pem', 'wb') as f:
+        f.write(crypto.dump_privatekey(crypto.FILETYPE_PEM, pkey))
+    with open('cert.pem', 'wb') as f:
+        f.write(crypto.dump_certificate(crypto.FILETYPE_PEM, cert))
+
+# all handler path entrypoint
+@authenticate
+async def handler(websocket, path):
+    logger.info(f"path = {path}")
+    if path.startswith("/upload"):
+        # Extract file name and directory from the path
+        dir_path = os.path.normpath(path[len("/upload/"):])
+        file_name = path.split("/")[-1]
+        directory = os.path.dirname(dir_path)
+        logger.info(f"file_name:{file_name}, file_path:{directory}")
+        await handle_upload(websocket, file_name, directory)
+    elif path.startswith("/cmd"):
+        # Extract command from the path
+        command = urllib.parse.unquote(path.split("/cmd")[1].strip("/"))
+        if command:
+            await handle_cmd(websocket, command)
+        else:
+            await websocket.send("No command specified")
+    elif path.startswith("/py"):
+        await handle_py_script(websocket)
+    elif path.startswith("/sh"):
+        await handle_shell_script(websocket)
+
+    else:
+        await websocket.send("Invalid path")
+
+# execute python script
+async def handle_py_script(websocket):
+    data = await websocket.recv()
+    with tempfile.NamedTemporaryFile(delete=True) as temp:
+        temp.write(data.encode('utf-8'))
+        temp.flush()  # flush data to disk
+        # get the file path and file name
+        file_path = temp.name
+        logger.info(file_path)
+        with subprocess.Popen(['python3', '-u', file_path],
+                              stdout=subprocess.PIPE,stderr=subprocess.STDOUT,
+                              bufsize=1,
+                              universal_newlines=True) as process:
+            for line in process.stdout:
+                line = line.rstrip()
+                logger.info(f"line = {line}")
+                await websocket.send(line)
+            # get the process return code
+            return_code = process.wait()
+            logger.info(f"Command executed with return code: {return_code}")
+        await websocket.close(code=1000, reason=f"{return_code}")
+
+# execute shell script
+async def handle_shell_script(websocket):
+    data = await websocket.recv()
+    return_code = -1
+    # create temp file and delete the file outside the with seq
+    with tempfile.NamedTemporaryFile(delete=True) as temp:
+        temp.write(data.encode('utf-8'))
+        temp.flush()  # flush data to disk
+        # get file name
+        file_path = temp.name
+        logger.info(file_path)
+        with subprocess.Popen(['bash', file_path],
+                              stdout=subprocess.PIPE,stderr=subprocess.STDOUT,
```

Comment:
Need to pass parameters to the script.The script execution case is as follows:
```bash kubelet_node_helper.sh join dns_address```
```bash kubelet_node_helper.sh logs kubelet```

Reply:
fixed. use client.py test it work fine, you should use the url path to pass the args